paper-tooltip.html

```diff
@@ -81,331 +72,413 @@
+               * played when showing the tooltip.
+               */
+              animationDelay: {
+                  type: Number,
+                  value: 500
+              },
+              /**
+               * This property is deprecated.  Use --paper-tooltip-animation to change the animation.
+               * The entry and exit animations that will be played when showing and
+               * hiding the tooltip. If you want to override this, you must ensure
+               * that your animationConfig has the exact format below.
+               * @deprecated since version
+               *
+               * The entry and exit animations that will be played when showing and
+               * hiding the tooltip. If you want to override this, you must ensure
+               * that your animationConfig has the exact format below.
+               */
+              animationConfig: {
+                  type: Object,
+                  value: function() {
+                      return {
+                          'entry': [{
+                              name: 'fade-in-animation',
+                              node: this,
+                              timing: {delay: 0}
+                          }],
+                          'exit': [{
+                              name: 'fade-out-animation',
+                              node: this
+                          }]
+                      }
+                  }
+              },
+              _showing: {
+                  type: Boolean,
+                  value: false
+              }
+          },
+          listeners: {
+              'webkitAnimationEnd': '_onAnimationEnd',
+          },
+          /**
+           * Returns the target element that this tooltip is anchored to. It is
+           * either the element given by the `for` attribute, or the immediate parent
+           * of the tooltip.
+           */
+          get target () {
+              var parentNode = Polymer.dom(this).parentNode;
+              // If the parentNode is a document fragment, then we need to use the host.
+              var ownerRoot = Polymer.dom(this).getOwnerRoot();
+              var target;
+              if (this.for) {
+                  target = Polymer.dom(ownerRoot).querySelector('#' + this.for);
+              } else {
+                  target = parentNode.nodeType == Node.DOCUMENT_FRAGMENT_NODE ?
+                      ownerRoot.host : parentNode;
+              }
+              return target;
+          },
+          attached: function() {
+              this._findTarget();
+          },
+          detached: function() {
+              if (!this.manualMode)
+                  this._removeListeners();
+          },
+          /**
+           * Replaces Neon-Animation playAnimation - just calls show and hide.
+           * @Deprecated Use show and hide instead.
+           * @param {string=} type
+           */
+          playAnimation: function(type) {
```

Comment:
Please add a `cancelAnimation` function. Some tests still failing because of this.

I believe it should have something along the lines of:

```js
cancelAnimation: function() {
  this.$.tooltip.classList.remove(this._getAnimationType('entry'));
  this.$.tooltip.classList.remove(this._getAnimationType('exit'));
},
```

I didn't have time to take a deep look at how this would affect internal state so you may want to iterate on my impl